homeassistant/components/lidarr/config_flow.py

```diff
@@ -0,0 +1,111 @@
+"""Config flow for Lidarr."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from aiohttp import ClientConnectorError
+from aiopyarr import SystemStatus, exceptions
+from aiopyarr.lidarr_client import LidarrClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow
+from homeassistant.const import CONF_API_KEY, CONF_URL, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_NAME, DOMAIN
+
+
+class LidarrConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Lidarr."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.entry: ConfigEntry | None = None
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        if user_input is not None:
+            return await self.async_step_user()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="reauth_confirm")
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            user_input = dict(self.entry.data) if self.entry else None
+
+        else:
+            try:
+                result = await validate_input(self.hass, user_input)
+                if isinstance(result, tuple):
+                    user_input[CONF_API_KEY] = result[1]
+                elif isinstance(result, str):
+                    errors = {"base": result}
```

Comment:
We shouldn't use a string directly from a 3rd party library as a translation string. I suggest raising an error in the validator function, catch that error here and set an appropriate error string.